lib/db.js

```diff
@@ -1,18 +1,43 @@
 var logger = require('juttle/lib/logger').getLogger('sql-db-init');
-
+var _ = require('underscore');
 var knex;
+var savedDbId;
+var confArr;
 
 var DB = {
     init: function(config) {
-        if (config.knex) {
-            knex = config.knex;
+        confArr = _.isArray(config) ? config : [config];
+    },
+    getDbConnection: function (options) {
+        options = options || {};
+
+        var dbid = `${options.id}-${options.db}`;
+        if (knex && dbid === savedDbId) {
+            return knex;
+        }
+
+        var conf = _.findWhere(confArr, {id: options.id}) || confArr[0];
+        knex = DB.getKnex(conf, options);
+        savedDbId = dbid;
+
+        try {
             logger.info('initializing db connection with conf:', knex.client.config);
```

Comment:
How can this throw an error, is `knex.client` undefined sometimes? That seems like a pretty roundabout validation, how about we just check explicitly `if (!knex.client) { throw new Error('blah blah blah'); }`?